addon_service/common/waterbutler_compat.py

```diff
@@ -1,51 +1,63 @@
+import dataclasses
+import enum
+
 from rest_framework_json_api import serializers
 
-from addon_toolkit import credentials
+from addon_service.common import known_imps
+from addon_service.common.enum_decorators import enum_names_same_as
+from addon_toolkit import (
+    AddonImp,
+    credentials,
+    json_arguments,
+)
+
+
+@enum_names_same_as(known_imps._KnownAddonImps)
+class WaterbutlerProviderKey(enum.Enum):
+    BOX_DOT_COM = "box"
+
+    if __debug__:
+        BLARG = "blarrrg"
+
+    @staticmethod
+    def for_imp_cls(imp_cls: type[AddonImp]) -> str:
+        _imp_name = known_imps.get_imp_name(imp_cls)
+        _provider_key = WaterbutlerProviderKey[_imp_name].value
+        if _provider_key is None:
```

Comment:
oh, that was to support imps that do not correspond to a waterbutler provider (at one point had `BLARG = None`, since the name is required to be there), but in that case it's probably better to return the `None` value 👍

Reply:
Oh, yeah, `same_names` means the KeyError won't happen. But this is making me question the `same_names` decorator, since even non-Storage Imps will need an entry here...I've got a couple thoughts on how to handle this while ensuring the static mapping is complete, but let's chat about that for another ticket.

Reply:
yeahhh for `WaterbutlerProviderKey` in particular, a partial mapping might be better... or could include the "imp name" and move that mapping into waterbutler... but yeah agree, future work